### PR TITLE
#1370 Add a switch to add the possibility to disable the call to xerc…

### PR DESCRIPTION
**Pull request by rosiereflo**
…es init and terminate methods which are not thread safe

### CODE REVIEW

**Comment by rosiereflo** [comment]
Compilation fails due to the fact that 3rd party are not recompiled in the coverage test

File: dynawo/3rdParty/libxml/sax/parser/include/xml/sax/parser/ParserFactory.h [hunk 1]
```diff
@@ -53,6 +53,8 @@ class ParserFactory {
    * @brief allocates a new parser.
    */
   Parser* newParser() const;
+private:
+  bool multiThreadingMode_;
```
Comment (lecourtoisflo) [inline dynawo/3rdParty/libxml/sax/parser/include/xml/sax/parser/ParserFactory.h:57]: Proposition: use a default value to false for this member. This leads to less modifications

**Review by gautierbureau — APPROVED** [review]

**Review by lecourtoisflo — CHANGES_REQUESTED** [review]

**Comment by gautierbureau** [comment]
no issue with libxml2 and the new iidm library ?

File: dynawo/3rdParty/libxml/sax/parser/include/xml/sax/parser/ParserFactory.h (hunk 1 above)
Comment (rosiereflo) [reply dynawo/3rdParty/libxml/sax/parser/include/xml/sax/parser/ParserFactory.h:57]: I did this on purpose to force the developer to think about whether this is called in multi-threading or not and to explicitely add the init and terminate of xerces in his application if  required

**Comment by rosiereflo** [comment]
> no issue with libxml2 and the new iidm library ?

I didn't see any, it will be tested extensively when dynawo 1.3.0 will be used ina cpp11 environment
.

**Comment by rosiereflo** [comment]
http://xmlsoft.org/threads.html

I guess we will have to follow those guidelines

**Review by lecourtoisflo — APPROVED** [review]

**Comment by rosiereflo** [comment]
@gautierbureau if everything is ok can we merge this branch? thanks

**Comment by rosiereflo** [comment]
I just added the modifications for libxml2

File: dynawo/sources/Modeler/DataInterface/PowSyblIIDM/DYNDataInterfaceIIDM.cpp [hunk 2]
```diff
@@ -82,12 +82,12 @@ using criteria::CriteriaCollection;
...
 class LibXml2 {
  public:
-  LibXml2() {
+  explicit LibXml2(bool multiThreadingMode) {
+    if (!multiThreadingMode)
       xmlInitParser();
   }
 
   ~LibXml2() {
-      xmlCleanupParser();
```
Comment (gautierbureau) [inline dynawo/sources/Modeler/DataInterface/PowSyblIIDM/DYNDataInterfaceIIDM.cpp:90]: why this delete ?

File: dynawo/sources/Modeler/DataInterface/PowSyblIIDM/DYNDataInterfaceIIDM.cpp [hunk 3]
```diff
@@ -135,7 +135,6 @@ DataInterfaceIIDM::build(std::string iidmFilePath) {
 void
 DataInterfaceIIDM::dumpToFile(const std::string& iidmFilePath) const {
   try {
-    LibXml2 libxml2;
```
Comment (gautierbureau) [inline dynawo/sources/Modeler/DataInterface/PowSyblIIDM/DYNDataInterfaceIIDM.cpp:138]: why this delete ?

File: dynawo/sources/Modeler/DataInterface/PowSyblIIDM/DYNDataInterfaceIIDM.cpp (hunk 2 above)
Comment (rosiereflo) [reply dynawo/sources/Modeler/DataInterface/PowSyblIIDM/DYNDataInterfaceIIDM.cpp:90]: xmlCleanupParser is not required, it is used only to remove false positive memory leak from valgrind.

File: dynawo/sources/Modeler/DataInterface/PowSyblIIDM/DYNDataInterfaceIIDM.cpp (hunk 3 above)
Comment (rosiereflo) [reply dynawo/sources/Modeler/DataInterface/PowSyblIIDM/DYNDataInterfaceIIDM.cpp:138]: not needed

**Review by gautierbureau — APPROVED** [review]

**Pull request closed without merging**